power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/input.hpp

```diff
@@ -21,36 +21,78 @@ struct BaseInput {
     ID id;  // ID of the object
 };
 
-struct NodeInput : BaseInput {
+static_assert(std::is_standard_layout_v<BaseInput>);
+
+struct NodeInput {
+    ID id;  // ID of the object
     double u_rated;  // rated line-line voltage
+
+    // implicit conversions to BaseInput
+    operator BaseInput&() { return reinterpret_cast<BaseInput&>(*this); }
+    operator BaseInput const&() const { return reinterpret_cast<BaseInput const&>(*this); }
```

Comment:
every implicit conversion introduces 2 new sonar cloud code smells. that's 4 per base type, and 250 new ones in total.

I'm personally not a fan of adding `// NOSONAR` to every single line.

I've also read that it's also possible to do `*static_cast<BaseInput*>(static_cast<void*>(this))`. if the alignment is the same, it should actually be possible to do directly `*static_cast<BaseInput*>(this)`, but for some reason that's not working here. see also https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1120 and https://godbolt.org/z/E884oTEqr

In addition, it also complains that we should use `std::bit_cast` when dealing with types that have no additional attributes (e.g. `SensorShortCircuitOutput`). In those cases, shouldn't we just use e.g. `using SensorShortCircuitOutput = BaseOutput` instead?

Reply:
Let's not make thing more difficult. We know what we are doing with `reintepret_cast` here. Just disable globally the warning from SonarCloud.